test/TestExtensions/API.Test/InternalAPITestHook.cs

```diff
@@ -87,5 +88,33 @@ public static bool ExecuteInitScript(string id, string version)
 
             return false;
         }
+
+        public static bool BatchEventsApi(string id, string version)
+        {
+            var dte = ServiceLocator.GetInstance<EnvDTE.DTE>();
+            var packageProjectEventService = ServiceLocator.GetInstance<IVsPackageInstallerProjectEvents>();
+            var installerServices = ServiceLocator.GetInstance<IVsPackageInstaller>();
+            var batchStartIds = new List<string>();
+            var batchEndIds = new List<string>();
+
+            packageProjectEventService.BatchStart += (o, args) =>
+            {
+                batchStartIds.Add(args.BatchId);
+            };
+
+            packageProjectEventService.BatchEnd += (o, args) =>
+            {
+                batchEndIds.Add(args.BatchId);
+            };
+
+            foreach (EnvDTE.Project project in dte.Solution.Projects)
+            {
+                installerServices.InstallPackage(null, project, id, version, false);
+            }
+
+            return batchStartIds.Count == 1 &&
+                   batchEndIds.Count == 1 &&
+                   batchStartIds[0].Equals(batchEndIds[0], StringComparison.OrdinalIgnoreCase);
```

Comment:
nit: ordinal compare, they are guids, but still the batches shouldn't randomly change case